### DISCUSSION TITLE
Unable to install CentOS 7 on M1

### DISCUSSION THREAD

**Discussion by JadedTuna** [utmapp/UTM#2659] [UTM Help (macOS)]
**Describe the issue**
I have a VM with ARM64 architecture on M1. I can boot it into GRUB (I see "Install CentOS 7", "Test media...", etc). However, selecting any option results in a black screen with _ in the top left corner, and nothing else for around a minute. Then the screen changes back to previous GRUB menu (e.g. if I entered the "Troubleshooting" menu first, and then selected some option, after the black screen installer returns to the "Troubleshooting" menu.

I have tried doing the same with Rocky Linux 8 (ARM64) with same results.

**Configuration**
* UTM Version: 2.1.1 (30)
* OS Version: 11.4
* Intel or Apple Silicon? Apple Silicon

**Crash log**
Not applicable.

**Debug log**
[debug.log](https://github.com/utmapp/UTM/files/6778042/debug.log)

**Upload VM**
[config.plist.txt](https://github.com/utmapp/UTM/files/6778066/config.plist.txt)

**Comment by Drallas** [comment]
Same with CentOS stream 8 today.. Did you get it to work?

**Comment by AwlsomeAlex** [comment]
@Drallas @JadedTuna Don't think their kernels (which are quite old) support M1. Fedora and CentOS Stream 9 Alpha (atm based on Fedora 34, what will be RHEL9) work fine so makes me imagine its got something to do with that.

**Reply by bergstenarn** [reply]
I was struggling with bringing up a Centos8 VM in UTM with the same disheartening boot problem as described in this discussion (MacBook Air m1). After I found this discussion I instead tried Centos9 ... and the VM booted up and Centos9 got successfully installed.

However I experienced problems with the mouse pointer as clicking didn't work. But after toggling Capture Mouse (CTRL+OPT keys) mouse clicks started to work.

And note that the final installation reboot will present the same _install or try_ Centos GRUB menu as when you started the installation. To work around that, you need to power off the Centos VM and then in UTM modify the boot order of the Centos VM to make _VirtIO Drive_ the first (topmost in the list) volume to try when booting. Select _VirtIO Drive_ and click Move Up. Now your Centos VM install should boot nicely.

<img width="147" alt="UTM_Centos9_disk_boot_order" src="https://user-images.githubusercontent.com/34752788/162163086-3d7c1f5a-57d1-44f9-b541-f9bf3479e65e.png">

For the ones interested, here is an explanation telling why Centos7/8 (and the corresponding RHEL releases) won't run on Macs with m1 ... unless you're willing to spend time rebuilding the Centos7/8 Linux kernel ... which I know absolutely nothing about: https://communities.vmware.com/t5/Fusion-for-Apple-Silicon-Tech/State-of-linux-VM-compatibility-with-ARM/m-p/2892673

**Reply by TBytes404** [reply]
great solution ,,  it worked!!
thnx

**Reply by 3DCoded** [reply]
I'm downloading CentOS 9 to try it out.

> @Drallas @JadedTuna Don't think their kernels (which are quite old) support M1. Fedora and CentOS Stream 9 Alpha (atm based on Fedora 34, what will be RHEL9) work fine so makes me imagine its got something to do with that.

 I also read somewhere that Rocky Linux 9 is undergoing a major kernel change so maybe Rocky Linux 9 will work on M1.

**Reply by 3DCoded** [reply]
It worked!! Thank you @bergstenarn!

**Reply by wmheid** [reply]
Thanks @bergstenarn!  I am using UTM 4.3.5 and was successful with your instructions.  Admittedly, I am from a Hyper-V world and couldn't find where to change the boot order, so I simply deleted the USB Drive and everything booted as expected.

Of note is that I had to set the VM to 4 Cores manually and set the RAM to 4 GB.  It seems to pause with a flashing cursor in a couple of places, but waiting patiently for a few seconds lets it boot on its own.

Thanks again!

**Comment by kmbgit** [comment]
CentOS 7 runs on my 2021 14-in MacBook Pro (Apple M1 Pro CPU) under UTM 4.1.5 (installed via `brew install qemu utm`) if I untick the "Use Hypervisor" option in Settings under QEMU.  (Leaving it ticked gives an error message "This 64 KB granular kernel is not supported by your CPU).  The system (selected by default) is QEMU 7.2 ARM Virtual Machine.

I used this image: http://mirrors.coreix.net/centos-altarch/7.9.2009/isos/aarch64/CentOS-7-aarch64-Minimal-2009.iso

**Reply by sunsky** [reply]
great, it works
